Release notes for 3.5.4 and 3.4.38 (#631)
opensciencegrid/docs: docs/release/3.4/release-3-4-38.md

OSG Software Release 3.4.38
===========================

**Release Date**: 2019-10-23
**Supported OS Versions:** EL7, EL6

Summary of changes
------------------

This release contains:

> -   [StashCache-Client 5.5.0](https://github.com/opensciencegrid/StashCache/releases/tag/v5.5.0): Can be installed as a Python package

Reviewer: This isn't a user-visible change. The previous version we shipped was 5.2.0 and there have been multiple releases since then. How about
```suggestion
-   [StashCache-Client 5.5.0](https://github.com/opensciencegrid/StashCache/releases/tag/v5.5.0): Bugfixes and improved SciTokens support
```

-   Updated CA certificates based on [IGTF 1.102](http://dist.eugridpma.info/distribution/igtf/current/CHANGES):
    -   Added CESNET-CA-4 ICA accredited classic CA for issuer roll-over (CZ)
-   [VO Package v97](https://github.com/opensciencegrid/osg-vo-config/releases/tag/release-97): Added CLAS12 VO for Jefferson Lab

These [JIRA tickets](https://jira.opensciencegrid.org/issues/?jql=project%20%3D%20SOFTWARE%20AND%20fixVersion%20%3D%203.4.38%20ORDER%20BY%20priority%20DESC%2C%20key%20DESC) were addressed in this release.

Containers
----------

The [Worker node containers](/worker-node/using-wn-containers/) have been updated to this release.

Notes
-----

This section describes important upgrade notes and/or caveats for packages available in the OSG release repositories.
Detailed changes are below. All of the documentation can be found [here](/index.md).

-   OSG 3.4 contains only 64-bit components.

> -   StashCache is only supported on EL7

Reviewer: ```suggestion
-   The StashCache service is only supported on EL7
```
(stashcache-client works just fine on EL6)

### GlideinWMS ###

1. GlideinWMS 3.4.6 is the last release supporting Globus GRAM (a.k.a. GT2/GT5).

1. For new Singularity features introduced in GlideinWMS 3.4.1, all factories and frontends need to be >= 3.4.1.

    !!! note
        OSG GlideinWMS factories are running at least 3.4.1

    If some of the connected Factories are <= 3.4.1 you will see an error during reconfig/upgrade if you try to use
    features that require a newer Factory.
    To start using Singularity via GlideinWMS, see:

    - <https://glideinwms.fnal.gov/doc.prd/frontend/configuration.html#singularity>
    - <https://glideinwms.fnal.gov/doc.prd/factory/configuration.html#singularity>
    - <https://glideinwms.fnal.gov/doc.prd/factory/custom_vars.html#singularity_vars>

> 1. Upgrades from <= 3.4.0 may require merging `/etc/condor/config.d/*.rpmnew` files and a restart of HTCondor.

Reviewer: ```suggestion
1. Upgrades from <= 3.4.0 may require merging `/etc/condor/config.d/*.rpmnew` files and restarting HTCondor.
```

1. GlideinWMS >= 3.4.5 uses shared port, requiring only port 9618.

>    To ease the transition to shared port, the User Collector secondary collectors and CCBs support both shared and

Reviewer: ```suggestion
   To ease the transition to shared port, the User Collector, secondary collectors, and CCBs support both shared and
```

   separate, individual ports.
   To start using shared port, change the secondary collectors lines and the CCBs lines (if any) in
   `/etc/gwms-frontend/frontend.xml`, changing the address to include the shared port sinful string:

        <collector DN="/DC=org/DC=opensciencegrid/O=Open Science Grid/OU=Services/CN=gwms-frontend.domain" group="default" node="gwms-frontend.domain:9618?sock=collector0-40" secondary="True"/>

>    Replacing `gwms-frontend-domain` with the hostname of your GlideinWMS frontend.

Reviewer: ```suggestion
   Replace `gwms-frontend-domain` above with the hostname of your GlideinWMS frontend.
```
